src/Selenium2Library/keywords/_screenshot.py

```diff
@@ -88,13 +124,25 @@ def _restore_screenshot_directory(self):
 
     def _get_screenshot_paths(self, filename):
         if not filename:
-            self._screenshot_index += 1
-            filename = 'selenium-screenshot-%d.png' % self._screenshot_index
+            filename = 'selenium-screenshot-{index}.png'
+            index = self._get_screenshot_index(filename)
+            filename = filename.format(index=index)
+        elif '{index' in filename:
+            index = self._get_screenshot_index(filename)
+            filename = filename.format(index=index)
+            filename = filename.replace('/', os.sep)
         else:
             filename = filename.replace('/', os.sep)
```

Comment:
There's quite a bit of duplication above. At least the `elif` could be changed to `if` and two lines above it removed, but perhaps the code could be simplified to this:

```python
    def _get_screenshot_paths(self, filename):
        if not filename:
            filename = 'selenium-screenshot-{index}.png'
        else:
            filename = filename.replace('/', os.sep)
        filename = filename.format(index=self._get_screenshot_index(filename))
```